### ISSUE TITLE
Events API

### ISSUE THREAD

**Issue by to-sta** [activist-org/activist#628]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

We are starting with creating the API's that will be used by the frontend in production soon 😃. 

We are using [django-rest-framework](https://www.django-rest-framework.org/) in the backend to create our API's. ViewSets in particular the [ModelViewSet](https://www.django-rest-framework.org/api-guide/viewsets/#modelviewset) are used for creating the different HTTP methods. Another benefit of using the ModelViewSet is that we are getting some validation from the model itself and only need to extend that in the serializer if necessary.

Have look at the [scheme](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=5377-96371&mode=design&t=8xoYzpakJ6ebt8Ie-4) and the models.py file. 

Task is to implement all these methods for the OrganizationViewSet that represent the different HTTP methods:

```python
def list(self, request):
    pass

def create(self, request):
    pass

def retrieve(self, request, pk=None):
    pass

def update(self, request, pk=None):
    pass

def partial_update(self, request, pk=None):
    pass

def destroy(self, request, pk=None):
    pass
```

The API generally is a combiniation of:

- views.py
- serializer.py
- models.py

Also feel free to suggest changes to the model, in case it is necessary for the logic. They are not set in stone 😃. In terms of testing, you can use Swagger UI for manuall testing (backend endpoint: `/v1/schema/swagger-ui/`).

>[!NOTE]
> Please use an appropiate http [status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) for your responses.

### Contribution

I am happy to help with this issue.

**Comment by to-sta** [comment]
Do you want to work on this issue @bharath637462 ?

**Comment by andrewtavis** [comment]
Yes @bharath637462, let's keep it to one API for now :)

**Comment by andrewtavis** [comment]
We can work on others as soon as some are finished.

**Comment by bharath637462** [comment]
yes @to-sta

**Comment by andrewtavis** [comment]
Let's please finish the work for the current frontend issue before starting on this one as well, @bharath637462.

**Comment by bharath637462** [comment]
@to-sta but event has already api written on it 
class EventViewSet(viewsets.ModelViewSet[Event]):
    queryset = Event.objects.all()
    serializer_class = EventSerializer
    pagination_class = CustomPagination
    throttle_classes = [AnonRateThrottle, UserRateThrottle]
    
    at backend/events/views.py
what shold i do could you please explain breifly

**Comment by andrewtavis** [comment]
@bharath637462, [this is the PR for resources](https://github.com/activist-org/activist/pull/625). That would roughly be what we're looking for :)

**Comment by UnknownSean8** [comment]
Hi @andrewtavis, was wondering if this issue is still open and what is needed to be done. I am looking at another issue #621 as well. Do let me know which issue you would like me to work on.

**Comment by andrewtavis** [comment]
Hey @UnknownSean8 👋 We're holding off on this issue for now. If you had thoughts on #621 then by all means write in there as we definitely need to think about how locations will be modeled on the platform. Your input would be invaluable! Also let me know on #527 as I mentioned most in the close of the other one you worked on :)

**Comment by UnknownSean8** [comment]
Ah okay, no problem, I am happy to work on #527. I saw that it's referencing #597 as well. Perhaps you could highlight the features/fixes that you would like me to work on so that I can better implement what you had in mind and assign me to it as well?

Thanks :D

**Comment by andrewtavis** [comment]
Thank you, @UnknownSean8! Do you want to write in there so I can assign and we can go from there with me outlining it more?

**Comment by josueemartinezz** [comment]
Hi @andrewtavis, are we still holding off on this issue?

**Comment by andrewtavis** [comment]
I think it'd be alright for you to give this a go, @josueemartinezz :) Let us know if you need any help here!
